src/wavm.cpp

```diff
@@ -393,4 +394,91 @@ ExecutionResult WavmEngine::internalExecute(
+    { "externalCodeCopy", IR::FunctionType{ {}, { IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32 } } },
+    { "getExternalCodeSize", IR::FunctionType{ { IR::ValueType::i32 }, { IR::ValueType::i32 } } },
+    { "getBlockCoinbase", IR::FunctionType{ {}, { IR::ValueType::i32 } } },
+    { "getBlockDifficulty", IR::FunctionType{ {}, { IR::ValueType::i32 } } },
+    { "getBlockGasLimit", IR::FunctionType{ { IR::ValueType::i64 }, {} } },
+    { "getTxGasPrice", IR::FunctionType{ {}, { IR::ValueType::i32 } } },
+    { "log", IR::FunctionType{ {}, { IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32 } } },
+    { "getBlockNumber", IR::FunctionType{ { IR::ValueType::i64 }, {} } },
+    { "getBlockTimestamp", IR::FunctionType{ { IR::ValueType::i64 }, {} } },
+    { "getTxOrigin", IR::FunctionType{ {}, { IR::ValueType::i32 } } },
+    { "storageStore", IR::FunctionType{ {}, { IR::ValueType::i32, IR::ValueType::i32 } } },
+    { "storageLoad", IR::FunctionType{ {}, { IR::ValueType::i32, IR::ValueType::i32 } } },
+    { "finish", IR::FunctionType{ {}, { IR::ValueType::i32, IR::ValueType::i32 } } },
+    { "revert", IR::FunctionType{ {}, { IR::ValueType::i32, IR::ValueType::i32 } } },
+    { "getReturnDataSize", IR::FunctionType{ { IR::ValueType::i32 }, {} } },
+    { "returnDataCopy", IR::FunctionType{ {}, { IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32 } } },
+    { "call", IR::FunctionType{ { IR::ValueType::i32 }, { IR::ValueType::i64, IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32 } } },
+    { "callCode", IR::FunctionType{ { IR::ValueType::i32 }, { IR::ValueType::i64, IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32 } } },
+    { "callDelegate", IR::FunctionType{ { IR::ValueType::i32 }, { IR::ValueType::i64, IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32 } } },
+    { "callStatic", IR::FunctionType{ { IR::ValueType::i32 }, { IR::ValueType::i64, IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32 } } },
+    { "create", IR::FunctionType{ { IR::ValueType::i32 }, { IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32, IR::ValueType::i32 } } },
+    { "selfDestruct", IR::FunctionType{ {}, { IR::ValueType::i32 } } }
+  };
+
+  for (auto const& import: moduleIR.functions.imports) {
+#if HERA_DEBUGGING
+    if (import.moduleName == "debug")
+      continue;
+#endif
+
+    ensureCondition(
+      import.moduleName == "ethereum",
+      ContractValidationFailure,
+      "Import from invalid namespace."
+    );
+
+    ensureCondition(
+      eei_signatures.count(import.exportName),
+      ContractValidationFailure,
+      "Importing invalid EEI method."
+    );
+    IR::FunctionType const& eei_function_type = eei_signatures.at(import.exportName);
+
+    ensureCondition(
+      moduleIR.types.size() >= import.type.index,
```

Comment:
@chfast is indexing of a vector going from 0 or 1? `moduleIR.types` is a vector.

Reply:
From 0.